synapse/handlers/e2e_keys.py

```diff
@@ -357,7 +357,7 @@ def _exception_to_failure(e):
     # include ConnectionRefused and other errors
     #
     # Note that some Exceptions (notably twisted's ResponseFailed etc) don't
-    # give a string for e.message, which json then fails to serialize.
+    # give a string for str(e), which json then fails to serialize.
     return {
```

Comment:
This might need extra handling like
errmsg = str(e)
if not isinstance(errmsg, six.string_types):
    errmsg="No error string provided"
return {